### PR TITLE
Require pin

### PR DESCRIPTION
**Pull request by djmdjm**

### CODE REVIEW

File: sk-usbhid.c [hunk 1]
```diff
@@ -564,19 +566,23 @@ sk_enroll(uint32_t alg, const uint8_t *challenge, size_t challenge_len,
...
-	if ((flags & SSH_SK_RESIDENT_KEY) != 0) {
+	if ((flags & (SSH_SK_RESIDENT_KEY|SSH_SK_USER_VERIFICATION_REQD)) != 0) {
 		if (check_sk_extensions(dev, "credProtect", &credprot) < 0) {
 			skdebug(__func__, "check_sk_extensions failed");
 			goto out;
 		}
 		if (credprot == 0) {
 			skdebug(__func__, "refusing to create unprotected "
-			    "resident key");
+			    "resident/pin-required key");
```
Comment (mfriedl) [inline sk-usbhid.c:576]: verification-required

File: ssh-keygen.c [hunk 2]
```diff
@@ -3511,6 +3532,8 @@ main(int argc, char **argv)
 		for (i = 0; i < nopts; i++) {
 			if (strcasecmp(opts[i], "no-touch-required") == 0) {
 				sk_flags &= ~SSH_SK_USER_PRESENCE_REQD;
+			} else if (strcasecmp(opts[i], "require-pin") == 0) {
```
Comment (mfriedl) [inline ssh-keygen.c:3535]: should this be user-verification-required, too?

File: auth-options.c [hunk 3]
```diff
@@ -569,13 +573,16 @@ sshauthopt_merge(const struct sshauthopt *primary,
...
+#define OPTFLAG_OR(x) ret->x = (primary->x == 1) || (additional->x == 1)
 	/* Permissive flags are logical-AND (i.e. must be set in both) */
 	OPTFLAG_AND(permit_port_forwarding_flag);
 	OPTFLAG_AND(permit_agent_forwarding_flag);
 	OPTFLAG_AND(permit_x11_forwarding_flag);
 	OPTFLAG_AND(permit_pty_flag);
 	OPTFLAG_AND(permit_user_rc);
 	OPTFLAG_AND(no_require_user_presence);
+	/* Restrictive flags are logical-OR (i.e. must be set in either) */
+	OPTFLAG_AND(require_verify);
```
Comment (mfriedl) [inline auth-options.c:585]: did you want to use _OR here?

File: sk-usbhid.c (hunk 1 above)
Comment (djmdjm) [reply sk-usbhid.c:576]: done

File: ssh-keygen.c (hunk 2 above)
Comment (djmdjm) [reply ssh-keygen.c:3535]: I made it verify-required to match auth-options.c

File: auth-options.c (hunk 3 above)
Comment (djmdjm) [reply auth-options.c:585]: oops!

**Comment by djmdjm** [comment]
I've made the changes you suggested and re-pushed. Please take another look!

**Comment by djmdjm** [comment]
(I also merged the manpage changes into the source commits, if you're wondering what happened to those)

**Pull request closed without merging**